File: java/src/test/java/com/alxarafe/domain/inventory/entity/BatchTest.java
package com.alxarafe.domain.inventory.entity;

import com.alxarafe.app.domain.catalogue.valueobject.ItemId;
import com.alxarafe.app.domain.inventory.entity.Batch;
import com.alxarafe.app.domain.inventory.valueobject.BatchCode;
import com.alxarafe.app.domain.inventory.valueobject.BatchId;
import org.junit.jupiter.api.Test;

import java.time.Instant;

import static org.junit.jupiter.api.Assertions.*;

class BatchTest {

    @Test
    void create() {
        var id = new BatchId("018e4e3a-3e7b-7b3e-8000-000000000001");
        var itemId = new ItemId("018e4e3a-3e7b-7b3e-8000-000000000002");
        var code = new BatchCode("BATCH-001");
        var expiration = Instant.parse("2027-01-01T00:00:00Z");
        var batch = new Batch(id, itemId, code, expiration);

        assertEquals(id, batch.id());
        assertEquals(itemId, batch.itemId());
        assertEquals(code, batch.batchCode());
        assertEquals(expiration, batch.expirationDate().orElseThrow());
    }

    @Test
    void createWithoutExpiration() {
        var batch = new Batch(
                new BatchId("018e4e3a-3e7b-7b3e-8000-000000000001"),
                new ItemId("018e4e3a-3e7b-7b3e-8000-000000000002"),
                new BatchCode("BATCH-001"),
                null);
        assertTrue(batch.expirationDate().isEmpty());
    }

    @Test
    void isExpired() {
        var batch = new Batch(
                new BatchId("018e4e3a-3e7b-7b3e-8000-000000000001"),
                new ItemId("018e4e3a-3e7b-7b3e-8000-000000000002"),
                new BatchCode("BATCH-001"),
                Instant.parse("2025-01-01T00:00:00Z"));
        assertTrue(batch.isExpired(Instant.parse("2026-01-01T00:00:00Z")));
    }

    @Test
    void isNotExpired() {
        var batch = new Batch(
                new BatchId("018e4e3a-3e7b-7b3e-8000-000000000001"),
                new ItemId("018e4e3a-3e7b-7b3e-8000-000000000002"),
                new BatchCode("BATCH-001"),
                Instant.parse("2027-01-01T00:00:00Z"));
        assertFalse(batch.isExpired(Instant.parse("2026-01-01T00:00:00Z")));
    }

    @Test
    void noExpirationNeverExpired() {
        var batch = new Batch(
                new BatchId("018e4e3a-3e7b-7b3e-8000-000000000001"),
                new ItemId("018e4e3a-3e7b-7b3e-8000-000000000002"),
                new BatchCode("BATCH-001"),
                null);
        assertFalse(batch.isExpired(Instant.parse("2030-01-01T00:00:00Z")));
    }
}

File: php/tests/Unit/Domain/Inventory/Entity/BatchTest.php
<?php

declare(strict_types=1);

namespace Tests\Unit\Domain\Inventory\Entity;

use Alxarafe\App\Domain\Catalogue\ValueObject\ItemId;
use Alxarafe\App\Domain\Inventory\Entity\Batch;
use Alxarafe\App\Domain\Inventory\ValueObject\BatchCode;
use Alxarafe\App\Domain\Inventory\ValueObject\BatchId;
use DateTimeImmutable;
use PHPUnit\Framework\TestCase;

final class BatchTest extends TestCase
{
    public function testCreate(): void
    {
        $id = new BatchId('018e4e3a-3e7b-7b3e-8000-000000000001');
        $itemId = new ItemId('018e4e3a-3e7b-7b3e-8000-000000000002');
        $code = new BatchCode('BATCH-001');
        $expiration = new DateTimeImmutable('2027-01-01');
        $batch = new Batch($id, $itemId, $code, $expiration);

        self::assertTrue($id->equals($batch->id()));
        self::assertTrue($itemId->equals($batch->itemId()));
        self::assertTrue($code->equals($batch->batchCode()));
        self::assertSame($expiration, $batch->expirationDate());
    }

    public function testCreateWithoutExpiration(): void
    {
        $batch = new Batch(
            new BatchId('018e4e3a-3e7b-7b3e-8000-000000000001'),
            new ItemId('018e4e3a-3e7b-7b3e-8000-000000000002'),
            new BatchCode('BATCH-001'),
            null,
        );
        self::assertNull($batch->expirationDate());
    }

    public function testIsExpired(): void
    {
        $batch = new Batch(
            new BatchId('018e4e3a-3e7b-7b3e-8000-000000000001'),
            new ItemId('018e4e3a-3e7b-7b3e-8000-000000000002'),
            new BatchCode('BATCH-001'),
            new DateTimeImmutable('2025-01-01'),
        );
        self::assertTrue($batch->isExpired(new DateTimeImmutable('2026-01-01')));
    }

    public function testIsNotExpired(): void
    {
        $batch = new Batch(
            new BatchId('018e4e3a-3e7b-7b3e-8000-000000000001'),
            new ItemId('018e4e3a-3e7b-7b3e-8000-000000000002'),
            new BatchCode('BATCH-001'),
            new DateTimeImmutable('2027-01-01'),
        );
        self::assertFalse($batch->isExpired(new DateTimeImmutable('2026-01-01')));
    }

    public function testNoExpirationNeverExpired(): void
    {
        $batch = new Batch(
            new BatchId('018e4e3a-3e7b-7b3e-8000-000000000001'),
            new ItemId('018e4e3a-3e7b-7b3e-8000-000000000002'),
            new BatchCode('BATCH-001'),
            null,
        );
        self::assertFalse($batch->isExpired(new DateTimeImmutable('2030-01-01')));
    }
}
